src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -54,17 +68,61 @@ public override void Dispose()
     private static readonly byte[] _jsonComma = [Convert.ToByte(',')];
     private static readonly byte[] _jsonClosingBracket = [Convert.ToByte(']')];
 
-    public override async Task ProcessAsync(ArraySegment<byte> data)
+    public override async Task ProcessAsync(ArraySegment<byte> data, CancellationToken cancellationToken)
     {
         IncrementBytesReceivedMetric(data.Count);
-        PipeReader request = PipeReader.Create(new ReadOnlySequence<byte>(data.Array!, data.Offset, data.Count));
+
+        IMemoryOwner<byte> memoryOwner = MemoryPool<byte>.Shared.Rent(data.Count);
+        data.AsSpan().CopyTo(memoryOwner.Memory.Span);
+        Memory<byte> memory = memoryOwner.Memory[..data.Count];
+
+        await _processChannel.Writer.WriteAsync(new ProcessRequest(memory, memoryOwner), cancellationToken);
+    }
+
+    public override async Task ReceiveLoopAsync(CancellationToken cancellationToken)
+    {
+        using AutoCancelTokenSource cts = cancellationToken.CreateChildTokenSource();
+
+        using ArrayPoolList<Task> allTasks = new(_workerTaskCount + 1);
+        allTasks.Add(Task.Factory.StartNew(async () =>
+        {
+            try
```

Comment:
Maybe throw in some thread names since live forever (till process end)
```suggestion
            Thread.CurrentThread.Name = "..."
            try
```